src/main/java/org/traccar/session/ConnectionManager.java

```diff
@@ -123,7 +123,9 @@ public DeviceSession getDeviceSession(
         Device device = deviceLookupService.lookup(uniqueIds);
 
         if (device == null && config.getBoolean(Keys.DATABASE_REGISTER_UNKNOWN)) {
-            device = addUnknownDevice(uniqueIds[0]);
+            if (uniqueIds[0].matches("[0-9]+") && !uniqueIds[0].isBlank()) {
```

Comment:
I think checking for `isBlank` is redundant here.

As for the regex, I think it's better to make it configurable. And for the default I would make it something like `\w{3,15}`. It would make it restrictive enough to filter the noise and generic enough to handle most of the identifiers.

Reply:
\w stands for "word character". Looks like it would match better with \d
\d{3,15} = matches any digit captures from 3 up to 15 digits
CMIIW

Reply:
Identifiers are not always all digits, so we need to make it more generic by default.

Reply:
"I think checking for isBlank is redundant here."

Please check this image
<img width="1273" alt="215695785-e0999f10-b206-4365-99c3-46be8100281f" src="https://user-images.githubusercontent.com/2969479/224240338-7ac37afe-ce0c-47b9-9438-398c9a591ca2.png">

There are one or more blank uniqueIds registered here

Reply:
```
if (device == null && config.getBoolean(Keys.DATABASE_REGISTER_UNKNOWN)) {
    if (uniqueIds[0].matches("\\w{3,15}")) {
        if (uniqueIds[0].matches("[0-9]+") && config.getBoolean(Keys.DATABASE_REGISTER_UNKNOWN_NUMBER_ONLY)) {
            device = addUnknownDevice(uniqueIds[0]);
        } else {
            device = addUnknownDevice(uniqueIds[0]);
        }
    }
}
```

Like this?

Reply:
A blank string doesn't match the regex, so how does it get added? Am I missing something?

Your code snippet doesn't look correct. I think the regex itself should be configurable. It shouldn't be a boolean config.

Reply:
```
if (device == null && config.getBoolean(Keys.DATABASE_REGISTER_UNKNOWN)) {
    if (uniqueIds[0].matches("\\w{3,15}")) {
        if (uniqueIds[0].matches(config.getString(Keys.DATABASE_REGISTER_UNKNOWN_REGEX))) {
            device = addUnknownDevice(uniqueIds[0]);
        } else {
            device = addUnknownDevice(uniqueIds[0]);
        }
    }
}
```

How it does like?

Reply:
No. Why do you have two regular expressions?

Reply:
> No. Why do you have two regular expressions?

```And for the default I would make it something like \w{3,15}```

I think it will be for default value, then if there are other can be added on config.

Reply:
Just like this?:
```
if (device == null && config.getBoolean(Keys.DATABASE_REGISTER_UNKNOWN)) {
    if (uniqueIds[0].matches(config.getString(Keys.DATABASE_REGISTER_UNKNOWN_REGEX))) {
        device = addUnknownDevice(uniqueIds[0]);
    } else {
        device = addUnknownDevice(uniqueIds[0]);
    }
}
```

Reply:
Why is there else statement?

Reply:
if (device == null && config.getBoolean(Keys.DATABASE_REGISTER_UNKNOWN)) {
    if (uniqueIds[0].matches(config.getString(Keys.DATABASE_REGISTER_UNKNOWN_REGEX))) {
        device = addUnknownDevice(uniqueIds[0]);
    }
}

Reply:
Yes, something like that.